Content.Server/Temperature/Systems/TemperatureSystem.cs

```diff
@@ -14,42 +18,70 @@ public class TemperatureSystem : EntitySystem
+
+        public float UpdateInterval = 1.0f;
+
+        private float _accumulatedFrametime = 0.0f;
+
         public override void Initialize()
         {
-            SubscribeLocalEvent<TemperatureComponent, OnTemperatureChangeEvent>(ChangeDamage);
+            SubscribeLocalEvent<TemperatureComponent, OnTemperatureChangeEvent>(EnqueueDamage);
             SubscribeLocalEvent<TemperatureComponent, AtmosExposedUpdateEvent>(OnAtmosExposedUpdate);
             SubscribeLocalEvent<ServerAlertsComponent, OnTemperatureChangeEvent>(ServerAlert);
+            SubscribeLocalEvent<TemperatureProtectionComponent, ModifyChangedTemperatureEvent>(OnTemperatureChangeAttempt);
         }
 
-        public void ForceChangeTemperature(EntityUid uid, float temp, TemperatureComponent? temperature = null)
+        public override void Update(float frameTime)
         {
-            if (Resolve(uid, ref temperature))
+            base.Update(frameTime);
+
+            _accumulatedFrametime += frameTime;
+
+            if (_accumulatedFrametime < UpdateInterval)
+                return;
+            _accumulatedFrametime -= UpdateInterval;
+
+            if (!ShouldUpdateDamage.Any())
+                return;
+
+            foreach (var comp in ShouldUpdateDamage)
             {
-                float lastTemp = temperature.CurrentTemperature;
-                float delta = temperature.CurrentTemperature - temp;
-                temperature.CurrentTemperature = temp;
-                RaiseLocalEvent(uid, new OnTemperatureChangeEvent(temperature.CurrentTemperature, lastTemp, delta));
+                if (comp.Deleted || comp.Paused)
+                    continue;
+
+                ChangeDamage(comp.OwnerUid, comp);
             }
+
+            ShouldUpdateDamage.Clear();
         }
 
-        public void ReceiveHeat(EntityUid uid, float heatAmount, TemperatureComponent? temperature = null)
+        public void ForceChangeTemperature(EntityUid uid, float temp, TemperatureComponent? temperature = null)
         {
             if (Resolve(uid, ref temperature))
             {
                 float lastTemp = temperature.CurrentTemperature;
-                temperature.CurrentTemperature += heatAmount / temperature.HeatCapacity;
-                float delta = temperature.CurrentTemperature - lastTemp;
-
+                float delta = temperature.CurrentTemperature - temp;
+                temperature.CurrentTemperature = temp;
                 RaiseLocalEvent(uid, new OnTemperatureChangeEvent(temperature.CurrentTemperature, lastTemp, delta));
             }
         }
 
-        public void RemoveHeat(EntityUid uid, float heatAmount, TemperatureComponent? temperature = null)
+        public void ChangeHeat(EntityUid uid, float heatAmount, TemperatureComponent? temperature = null)
         {
             if (Resolve(uid, ref temperature))
             {
+                var ev = new ModifyChangedTemperatureEvent(heatAmount);
```

Comment:
I think `ChangeHeat` should probably have some sort of `IgnoreHeatArmour` option.

A hardsuit should protect you from external heat, but it shouldn't prevent heat changes from metabolism or shivering.

Reply:
yeah that should be fairly easy to add